File: clients/shared_library/interfaces/post_application.ts
import { CreateApplicationAnswerMultiSelect } from './application_answer_multi_select'
import { CreateApplicationAnswerText } from './application_answer_text'
import { Student } from './student'

export interface PostApplication {
  student: Student
  answers_text: CreateApplicationAnswerText[]
  answers_multi_select: CreateApplicationAnswerMultiSelect[]
}

File: servers/core/applicationAdministration/applicationDTO/post_application.go
package applicationDTO

import (
	"github.com/ls1intum/prompt2/servers/core/student/studentDTO"
)

type PostApplication struct {
	Student            studentDTO.CreateStudent    `json:"student"` // should be able to handle either a new student or an existing dependent on ID
	AnswersText        []CreateAnswerText          `json:"answersText"`
	AnswersMultiSelect []CreateAnswerMultiSelect   `json:"answersMultiSelect"`
	AnswersFileUpload  []CreateAnswerFileUpload    `json:"answersFileUpload"`
}
